MerlinAU.sh

```diff
@@ -4642,83 +4968,535 @@ _Set_FW_UpdateCronSchedule_()
+              ;;
+           2) if _Set_FW_UpdateCronScheduleCustom_
+              then break ; fi
+              ;;
+           *) echo ; _InvalidMenuSelection_
+           ;;
+       esac
+
+       "$doReturnToMenu" && break
+   done
+}
+
+##----------------------------------------##
+## Modified by Martinski W. [2024-Nov-25] ##
+##----------------------------------------##
+_Toggle_ScriptAutoUpdate_Config_()
+{
+    local currentSetting  scriptUpdateCronSched  cronSchedStrHR
+    local keepOptionDisabled=false  retCode=1
 
     currentSetting="$(Get_Custom_Setting "Allow_Script_Auto_Up")"
 
-    if [ "$currentSetting" = "DISABLED" ]; then
-        printf "\n${REDct}*WARNING*${NOct}\n"
-        printf "Enabling this feature allows MerlinAU to self-update automatically without user action when a newer version becomes available.\n"
-        printf "Proceed at your own risk. This means both the script and the firmware become fully automatic.\n"
-        printf "The recommendation is always to read the changelogs on SNBForums or Github.\n"
-
-        if _WaitForYESorNO_ "\nProceed to ${REDct}ENABLE${NOct}?"; then
-            Update_Custom_Settings "Allow_Script_Auto_Up" "ENABLED"
-            printf "MerlinAU automatic script updates are now ${REDct}ENABLED.${NOct}\n"
-
-            # Validate the cron schedule
-            if _ValidateCronJobSchedule_ "$ScriptAU_CRON_DefaultSchedule"; then
-                # Add the cron job for automatic script updates
-                printf "Adding '${GRNct}${SCRIPT_UP_CRON_JOB_TAG}${NOct}' cron job for script updates...\n"
-                if _AddScriptAUCronJob_ ; then
-                    retCode=0
-                    printf "Cron job '${GRNct}${SCRIPT_UP_CRON_JOB_TAG}${NOct}' was added successfully.\n"
-                    current_schedule_english="$(translate_schedule "$ScriptAU_CRON_DefaultSchedule")"
-                    printf "Job Schedule: ${GRNct}${current_schedule_english}${NOct}\n"
-                    _AddScriptAutoUpdateHook_
+    if [ "$currentSetting" = "DISABLED" ]
+    then
+        printf "\n${REDct}*NOTICE*${NOct}\n"
+        printf "Enabling this feature allows the MerlinAU script to self-update automatically\n"
+        printf "without user action when a newer version becomes available. This means both the\n"
+        printf "script and the firmware updates become fully automatic. Proceed with caution.\n"
+        printf "The recommendation is to always read the changelogs on SNBForums or Github.\n"
+
+        if _WaitForYESorNO_ "\nProceed to ${MAGENTAct}ENABLE${NOct}?"
+        then
+            scriptUpdateCronSched="$(_GetScriptAutoUpdateCronSchedule_)"
+            cronSchedStrHR="$(_TranslateCronSchedHR_ "$scriptUpdateCronSched")"
+            printf "\nCurrent Schedule: ${GRNct}${scriptUpdateCronSched}${NOct}\n"
+            printf "[${GRNct}${cronSchedStrHR}${NOct}]\n"
+
+            if _WaitForYESorNO_ "\nConfirm daily check for automatic script updates?"
```

Comment:
I would change the wording to this. I wasn't sure what it was asking me at first.

![image](https://github.com/user-attachments/assets/d32c5175-f43a-4183-a05f-4d10564bb36b)


The "Proceed to enable" to me means it's being enabled, but really it seems to be the "Confirm daily check for automatic script updates?" that enables it. But it's asking that when it's showing the update schedule.

I would have it say: "Confirm the above schedule is accurate"

Reply:
> I would change the wording to this. I wasn't sure what it was asking me at first.
> ...
> 
> The "Proceed to enable" to me means it's being enabled, but really it seems to be the "Confirm daily check for automatic script updates?" that enables it. But it's asking that when it's showing the update schedule.
> 
> I would have it say: "Please confirm the above schedule for script updates looks correct"
>
>
The intent of the 2nd prompt is for users to **confirm** that they want to enable the feature **with** a **daily** check using the cron schedule being displayed, giving them further context so they are more informed of the choice they're making. If they answer "NO" to the confirmation, it means they are canceling their initial request to enable.

I'll see if I can re-word the message to make it clearer.

Reply:
@Martinski4GitHub 

Another choice is remove my original "proceed to enable?" So it goes straight to your validation of the schedule to be enabled then, would help cause less confusion I think?

Any suggestions? A reword could work too, or we could remove one of the "confirm, double confirm" prompts.